state/state.go

```diff
@@ -188,10 +188,17 @@ func StateFromProto(pb *tmstate.State) (*State, error) { //nolint:golint
 	state.ChainID = pb.ChainID
 	state.InitialHeight = pb.InitialHeight
 
-	bi, err := types.BlockIDFromProto(&pb.LastBlockID)
-	if err != nil {
-		return nil, err
+	var bi *types.BlockID
+	if pb.LastBlockID.DataAvailabilityHeader != nil {
+		var err error
+		bi, err = types.BlockIDFromProto(&pb.LastBlockID)
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		bi = &types.BlockID{}
 	}
```

Comment:
There are quite few errors reading empty BlockIDs from the dataStore, and so far, this is solution I've been using to patch them up. I think (hope) this is the main source of many of the remaining bugs.

Reply:
They're also getting silenced in some cases, which is leading to timeouts.